### ISSUE TITLE
False positive CA1805 when using parameterless value type ctor (C# 10)

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn-analyzers#5750]
### Analyzer

**Diagnostic ID**: [CA1805](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1805): `Do not initialize unnecessarily.`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest)

### Describe the bug

False positive when using parameterless value type ctor (C# 10).

### Steps To Reproduce
```C#
struct S
{
    public static readonly S Value = new(); //false positive CA1805

    public S() => throw null;
}
```

Beware that ctor does not need to be explicitly stated. In following example parameterless ctor is generated implicitly and CA1805 keeps to be false positive.

```C#
struct S
{
    public static readonly S Value = new(); //false positive CA1805

    private List<int> list = new();
}
```

**Comment by Evangelink** [comment]
Thanks for the report @kindermannhubert. I totally missed this change of behavior and was about to tell you you were wrong but as far as I can see from https://docs.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-10.0/parameterless-struct-constructors#new the behavior has indeed changed.

@mavasani I am trying to add a reproducing test and it seems that even when setting the `LanguageVersion` to `C# 10` I can't create a `struct` with the constructor:
```
// /0/Test0.cs(6,12): error CS0568: Structs cannot contain explicit parameterless constructors
```

**Comment by Youssef1313** [comment]
@Evangelink You might want to update `MicrosoftCodeAnalysisVersionForTests`

**Comment by Evangelink** [comment]
Thanks @Youssef1313!

**Comment by Evangelink** [comment]
Looks like your example is actually not a False Positive according to @stephentoub [answer here](https://github.com/dotnet/roslyn-analyzers/pull/5813/files#r833097672) but there is indeed a False Positive with instance members of structs.
